### PR TITLE
fix(cli): reset slash-command conflict dedupe when conflicts reappear

### LINKED ISSUE #24333: fix(cli): reset slash-command conflict dedupe when conflicts reappear
## Summary
`SlashCommandConflictHandler` permanently records every conflict key it has ever notified in `notifiedConflicts`, but it never expires or resets those entries when a conflict disappears and later reappears.

## Evidence
In `packages/cli/src/services/SlashCommandConflictHandler.ts`, `notifiedConflicts` is a process-lifetime `Set<string>`. `handleConflicts(...)` skips any conflict key that has ever been seen before, even if that conflict was resolved earlier in the session.

## Impact
If an extension, workspace command, or MCP prompt conflict is resolved and then reintroduced later, the user may never receive a second notification. The set also grows monotonically over the life of the process.

## Expected behavior
Users should be notified again when a previously-resolved conflict reappears, and the dedupe state should track currently-active conflicts rather than every historical conflict forever.

## Suggested fix
Track active conflict keys per flush or per current conflict state instead of using an unbounded permanent set, and add a regression test for reintroduced conflicts.

### PR DESCRIPTION
**Pull request by pmenic**
Closes #24333.

## Summary

`SlashCommandConflictHandler.notifiedConflicts` was a process-lifetime `Set<string>` that never expired entries. When a conflict was resolved and later reintroduced, the handler still treated it as already notified and silently dropped the feedback. The set also grew monotonically for the life of the process.

## Fix

The handler now treats each `SlashCommandConflictsPayload` as the current snapshot of active conflicts:

- Conflicts present in the payload that were already notified stay deduped, so a continuously active conflict is reported once, not on every reload.
- Conflicts not in the payload are dropped from `notifiedConflicts`, so a reappearance after resolution fires a fresh notification.
- Pending notifications are filtered against each new snapshot, so a conflict that resolves and reappears before the debounce flush is emitted once instead of producing stale or duplicate feedback.
- The set is replaced wholesale on each call, so its size tracks currently active conflicts rather than lifetime total conflicts.

`CommandService.create` now emits the conflict event unconditionally, including with an empty list. Without this, a reload that resolves all conflicts would leave the handler dedupe state holding stale keys, because the empty snapshot would never reach it. Empty payloads are a no-op for user feedback, so the behavior change is the cleared dedupe state.

## Tests

- Re-notifies when a previously resolved conflict reappears.
- Re-notifies only conflicts that were resolved and reappeared.
- Debounces newly active conflicts using snapshot semantics.
- Does not duplicate pending notifications when a conflict resolves and reappears before flush.

## Test plan

- [x] `npx vitest run packages/cli/src/services/SlashCommandConflictHandler.test.ts packages/cli/src/services/CommandService.test.ts`
- [x] `npx tsc --noEmit -p packages/cli/tsconfig.json`
- [x] From `packages/cli`: `npx vitest run src/services`

### CODE REVIEW

**Comment by pmenic** [comment]
Friendly ping — this PR has been waiting on the fork-PR `action_required` workflow approval since the last `main` merge on 2026-04-29. CLA is signed, no force-pushes, no rebases since the last review pass. Could a maintainer kick the workflow run when convenient? Thanks!

File: packages/cli/src/services/SlashCommandConflictHandler.ts [hunk 1]
```diff
@@ -40,24 +46,38 @@ export class SlashCommandConflictHandler {
...
-        c.loserExtensionName || c.loserMcpServerName || c.loserKind;
-      const key = `${c.name}:${sourceId}:${c.renamedTo}`;
-      if (this.notifiedConflicts.has(key)) {
-        return false;
+    const activeKeys = new Set<string>();
+    const newConflicts: SlashCommandConflict[] = [];
+
+    for (const c of payload.conflicts) {
+      const key = this.getConflictKey(c);
+      activeKeys.add(key);
```
Comment (spencer426) [inline packages/cli/src/services/SlashCommandConflictHandler.ts:54]: Update the loop to deduplicate against activeKeys as well. Add a test case specifically for intra-payload duplicates to verify the fix for the regression.

**Review by spencer426 — CHANGES_REQUESTED** [review]

**Comment by spencer426** [comment]
Thank you for your interest in contributing to the project! We are closing this PR due to inactivity.
